apps/evm_family/evm_typed_data_helper.c

```diff
@@ -187,9 +194,17 @@ bool evm_get_typed_struct_data_digest(
          sizeof(ETH_SIGN_TYPED_DATA_IDENTIFIER) - 1);
 
   offset += sizeof(ETH_SIGN_TYPED_DATA_IDENTIFIER) - 1;
-  eip712_status = hash_struct(&(typed_data->domain), data + offset);
+  if (0 < typed_data->domain.size) {
+    eip712_status = hash_struct(&(typed_data->domain), data + offset);
+  } else {
+    eip712_status = 0;
+  }
   offset += HASH_SIZE;
```

Comment:
please confirm if this increment should happen conditionally or not. this might result in an invalid final digest.